3-enrich/scala-common-enrich/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/adapters/registry/IgluAdapter.scala

```diff
@@ -47,11 +48,73 @@ object IgluAdapter extends Adapter {
 
   // Tracker version for an Iglu-compatible webhook
   private val TrackerVersion = "com.snowplowanalytics.iglu-v1"
+  private val VendorName = "Iglu"
 
   // Create a simple formatter function
   private val IgluFormatter: FormatterFunc = buildFormatter() // For defaults
 
   /**
+   * Attempts to parse a json string into a JValue
+   * example: {"p":"app"} becomes JObject(List((p,JString(app))))
+   *
+   * @param jsonStr The string we want to parse into a JValue
+   * @return a Validated JValue or a NonEmptyList Failure
+   *         containing a JsonParseException
+   */
+  private[registry] def parseJson(jsonStr: String): Validated[JValue] =
+    try {
+      parse(jsonStr).successNel
+    } catch {
+      case e: JsonParseException => {
+        val exception = JU.stripInstanceEtc(e.toString).orNull
+        s"${VendorName} event failed to parse into JSON: [${exception}]".failNel
+      }
+    }
+
+    /**
+     *
+     * Converts a payload into a list of validated events
+     * Expects a valid json - returns a single failure if one is not present
+     *
+     * @param body json payload as POST'd by sendgrid
+     * @param payload the rest of the payload details
+     * @return a list of validated events, successes will be the corresponding raw events
+     *         failures will contain a non empty list of the reason(s) for the particular event failing
+     */
+    private def payloadBodyToEvents(body: String, payload: CollectorPayload): List[Validated[RawEvent]] = {
+      try {
+
+        val parsed = parse(body)
+        val queryString = toMap(payload.querystring)
+        val schema = queryString.get("schema") match {
+          case Some(schemaUri) => schemaUri
+        }
+
+        if (parsed.children.isEmpty) {
```

Comment:
I guess I was thrown by it returning a List of RawEvents rather than a single object initially, as there is not actually any processing here to handle multiple events as of yet..